### PR TITLE
[7.x] Add str words count function

### PR DESCRIPTION
**Pull request by happytodev**
The purpose is to add an helper to count words in a string.

$jdoe = 'My name is John Doe';
echo Str::wordsCount($jdoe);

// display 5

### CODE REVIEW

**Comment by itanea** [comment]
I've updated documentation to : https://github.com/laravel/docs/pull/5876
Hope is this good format ;-)

**Comment by browner12** [comment]
Why do we need this method if it's just a wrapper for the PHP native `str_word_count`?

Also, you lose a little bit with this because you can longer pass the optional parameters to `str_word_count`.

**Comment by itanea** [comment]
> Why do we need this method if it's just a wrapper for the PHP native `str_word_count`?
> 
> Also, you lose a little bit with this because you can longer pass the optional parameters to `str_word_count`.

I understand your position. There is others methods who wrap natives PHP function like mb_substr or ucfirst. Then, I don't think this could be a problem.

But your second point is true. So I will modify to take care of native options of str_word_count.
I will do this today or tomorrow.

Thanks for your comment.

**Comment by GrahamCampbell** [comment]
Does the native function handle multibyte strings properly? Also, you will need to add tests and update the `Stringish` class too.

File: src/Illuminate/Support/Str.php [hunk 1]
```diff
@@ -712,4 +712,17 @@ public static function createUuidsNormally()
     {
         static::$uuidFactory = null;
     }
+
+    /**
+     * Count word(s) number in a string.
+     *
+     * @param  string  $value
+     * @param  int  $format
```
Comment (GrahamCampbell) [inline src/Illuminate/Support/Str.php]: `int|null`

File: src/Illuminate/Support/Str.php [hunk 2]
```diff
@@ -712,4 +712,17 @@ public static function createUuidsNormally()
...
+     * @param  string  $charlist
```
Comment (GrahamCampbell) [inline src/Illuminate/Support/Str.php]: `string|null`

File: src/Illuminate/Support/Str.php [hunk 3]
```diff
@@ -712,4 +712,17 @@ public static function createUuidsNormally()
...
+     * @return int
+     */
+    public static function wordsCount($value)
```
Comment (GrahamCampbell) [inline src/Illuminate/Support/Str.php]: Missing parameters.

File: src/Illuminate/Support/Str.php [hunk 4]
```diff
@@ -712,4 +712,17 @@ public static function createUuidsNormally()
...
+    {
+        return str_word_count($value, $format = null, $charlist = null);
```
Comment (GrahamCampbell) [inline src/Illuminate/Support/Str.php]: Incorrect null assignment.

**Review by GrahamCampbell — CHANGES_REQUESTED** [review]

File: src/Illuminate/Support/Str.php (hunk 4 above)
Comment (itanea) [reply src/Illuminate/Support/Str.php]: Oups, I'm tired :-(.
I will fix :D

**Review by happytodev — COMMENTED** [review]
I already fix this. Hope it's good.

**Review by GrahamCampbell — CHANGES_REQUESTED** [review]
Please also update the `Stringish` class, and add tests for both implementations.

**Comment by itanea** [comment]
> Please also update the `Stringish` class, and add tests for both implementations.

Sorry, but could you give some more details please ? I didn't found `Stringish` class, Maybe I didn't look hard enough.

Could you give me, please, more details ?

**Comment by mpyw** [comment]
He probably means `Stringable`

**Comment by itanea** [comment]
> He probably means `Stringable`

That's what I started looking at. I understand this is to add fluent string operation for this new method.
Ok, I will do it.

**Comment by itanea** [comment]
> Please also update the `Stringish` class, and add tests for both implementations.

Hello @GrahamCampbell I made changes.
For the Stringable class, I assumed that only the word count made sense. I did not implement the 'format' and 'charlist' options. 

What do you think of that?

**Comment by itanea** [comment]
> He probably means `Stringable`

Tanks to you ;-)

**Comment by GrahamCampbell** [comment]
Could you add a rest for the empty string case and a string that just contains a bunch of whitespace?

**Comment by itanea** [comment]
> Could you add a rest for the empty string case and a string that just contains a bunch of whitespace?

It's done ! ;-)

**Comment by itanea** [comment]
> Could you add a rest for the empty string case and a string that just contains a bunch of whitespace?

It's done ! ;-)

**Comment by taylorotwell** [comment]
Thanks for your pull request to Laravel!

Unfortunately, I'm going to delay merging this code for now. To preserve our ability to adequately maintain the framework, we need to be very careful regarding the amount of code we include. 

If possible, please consider releasing your code as a package so that the community can still take advantage of your contributions! 

If you feel absolutely certain that this code corrects a bug in the framework, please "**@**" mention me in a follow-up comment with further explanation so that GitHub will send me a notification of your response.

**Comment by itanea** [comment]
Thanks  @GrahamCampbell for your help.
I understand Taylor point of view. Just a little question, does this mean that it will never be implemented, or does it mean that it will eventually be postponed?

**Pull request closed without merging**